### DISCUSSION TITLE
Run TurboRepo Across Multiple Terminal Tabs

### DISCUSSION THREAD

**Discussion by amunrarara** [vercel/turborepo#7486] [Ideas]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

**Include an option to run a `turbo` pipeline, then separate each workspace's output into its own terminal window/tab.**

When TurboRepo runs, it spins up a pipeline consisting of several workspace processes and emits the log details to a single terminal window/tab. This can make process-specific debugging a bit of a nightmare, as we must slog through logs for multiple processes that are side-by-side in the same terminal instance. TurboRepo _does_ color-code the process' workspaces different from one another, which is very helpful, but it would seem a better solution to Include an option to run a `turbo` pipeline, then separate each workspace's output into its own terminal window/tab.

### Describe the solution you'd like

Include an option in turbo.json option to separate a turbo pipeline's workspaces into several terminal windows/tabs.

### Describe alternatives you've considered

Alternative solutions would bypass turbo completely. 

- Write a bash script that accepts a script command, then creates a new terminal window for each workspace in the pipeline, and runs '{yarn or alternative} workspace {command}' in that window.

- Manually open a terminal window for each workspace in the pipeline, then run the script within it.

**Comment by anthonyshew** [comment]
I'm wondering if a more approachable solution would be something like `turbo run dev --logs-filter=web`, where `logs-filter` would be the workspace that you are interested in seeing logs for.

**Comment by amunrarara** [comment]
@anthonyshew This would also be a great solution to add to the CLI, though it does not address the use case directly I'm hoping for, since I want to see logs for most or all of my workspaces still, but I want them to be separated for clarity. 

`--logs-filter` is very useful, though; is that already part of TurboRepo CLI?

**Comment by anthonyshew** [comment]
Ah, I see! I didn't quite put together that you still needed to see the logs from your various apps in parallel. I believe something like this would be new territory for `turbo`. I'm not sure that we have any existing functionality that moves across different shells. 🤔 

`--logs-filter` (or similar) is not an existing part of the CLI at this time.

**Comment by amunrarara** [comment]
It would be pretty awesome! A `--command` that runs a script to open up several shell instances, then output Turbo's logs among them, would be pretty dang nifty. I'm stuck bypassing Turbo completely otherwise if I want to separate logs during development, even for a single service. 

`--logs-filter` would be pretty nifty, too!

**Comment by amunrarara** [comment]
Maybe terminal multiplexing could play a part. I'll poke through the TurboRepo source this week and see if I can muster up a solution

**Comment by pixelpax** [comment]
Any updates here ?

**Comment by SirWrexes** [comment]
I'm using Expo and, currently, when ran through a Turbo pipeline, all control over the Expo server is lost as it gets ran in non-interactive mode.
The idea of spawning a new shell with the pipeline's elements in each their own would be great, as it could potentially mean gaining the interactivity with the Expo server back.
Same goes for Jest in watch mode.
It would truly be a killer feature !

**Comment by anthonyshew** [comment]
We've released a new terminal UI in 2.0 that should address the conversation here!

**Reply by stephen776** [reply]
Can you provide more info on how to accomplish this. I can't seem to find where to look in the docs. I am running a node backend and an expo app and would like the output in 2 separate terminal windows or something similiar

**Reply by stephen776** [reply]
Nevermind. Figured it out. Setting in turbo.json:

`”ui”:”tui”`
